src/instance-lookup.js

```diff
@@ -52,7 +73,23 @@ function instanceLookup(server, instanceName, callback, timeout, retries) {
       socket = dgram.createSocket('udp4');
       socket.on('error', onError);
       socket.on('message', onMessage);
-      socket.send(request, 0, request.length, SQL_SERVER_BROWSER_PORT, server);
+
+      if (multiSubnetFailover) {
```

Comment:
> Also we're doing lookupAll twice for instance-lookup code path. Once in instanceLookup() method and then again in connectOnPort(). We should only need to do this once I think. One way to do that would to invoke lookupAll in connect() method for all cases and then pass addresses as a parameter for connectOnPort() method as well as instance-lookup().

I'd expect that the server browser response contains IP-addresses, not DNS host names, so the connection would be done without another DNS lookup.